Realm/RLMClassInfo.mm

```diff
@@ -76,6 +76,20 @@
     return objectSchema->persisted_properties[property.index].column_key;
 }
 
+realm::ColKey RLMClassInfo::computedTableColumn(RLMProperty *property) const {
+    // Retrieve the table key and class info for the origin property
+    // that corresponds to the target property.
+    RLMClassInfo& originInfo = realm->_info[property.objectClassName];
+    TableKey originTableKey = originInfo.objectSchema->table_key;
+
+    TableRef originTable = realm.group.get_table(originTableKey);
+    // Get the column key for origin's forward link that links to the property on the target.
+    ColKey forwardLinkKey = originInfo.tableColumn(property.linkOriginPropertyName);
+
+    // The column key opposite of the origin's forward link is the target's backlink property.
+    return originTable->get_opposite_column(forwardLinkKey);
```

Comment:
I think this could be just:

```
    RLMClassInfo& originInfo = realm->_info[property.objectClassName];
    return originInfo.tableColumn(property.linkOriginPropertyName);
```

Reply:
The goal of the function is to return the colKey for the backlink (ie. computed) property on the _target_. I believe `originInfo.tableColumn(property.linkOriginPropertyName);` would be returning the colKey on the _origin_ (persisted) which points to the backlinked property.

So I think we'd still need to get the opposite column. 

Ultimately the return value for the method is used here:
```
static KeyPath KeyPathFromString(RLMRealm *realm,
                          RLMSchema *schema,
                          RLMClassInfo *info,
                          RLMObjectSchema *rlmObjectSchema,
                          NSString *keyPath) {
    RLMProperty *property;
    KeyPath keyPairs;

    NSArray<NSString *> *keyPathComponents = [keyPath componentsSeparatedByString:@"."];
    for (NSString *component in keyPathComponents) {
        property = rlmObjectSchema[component];
        RLMPrecondition(property, @"Invalid property name",
                        @"Property '%@' not found in object of type '%@'",
                        component, rlmObjectSchema.className);

        TableKey tk = info->objectSchema->table_key;
        ColKey ck;
        if (property.type == RLMPropertyTypeObject) {
            ck = info->tableColumn(property.columnName);
            info = &realm->_info[property.objectClassName];
            rlmObjectSchema = schema[property.objectClassName];
        } else if (property.type == RLMPropertyTypeLinkingObjects) {
    >>> ck = info->computedTableColumn(property); <<<
            info = &realm->_info[property.objectClassName];
            rlmObjectSchema = schema[property.objectClassName];
```
Returning just `originInfo.tableColumn(property.linkOriginPropertyName);` means that colKey is not found on the target (because it's the origin's property), and an exception is thrown.

Reply:
Ah, that is correct. I'd originally written the design assuming that we couldn't get the ColKey for a backlink and did everything in terms of the forward link keys and missed that core is expecting the backlink colkey here (which is better than what I had in mind).